### PR TITLE
Fixed: multi line chart does not draw line for first point on additional topics

### PR DESCRIPTION
**Pull request by gayanSandamal**
## Description

https://github.com/user-attachments/assets/5d30db67-689d-4298-b7c3-09cf0e7cf05c

## Related Issue(s)

https://github.com/FlowFuse/node-red-dashboard/issues/1255

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: ui/src/widgets/ui-chart/UIChart.vue [hunk 1]
```diff
@@ -383,8 +379,8 @@ export default {
 
                 this.chart.data.datasets.push(d)
             } else {
-                // have we seen this x-value before?
-                if (xIndex >= 0 && (this.props.xAxisType === 'category' || this.props.xAxisType === 'radial')) {
```
Comment (Steve-Mcl) [inline ui/src/widgets/ui-chart/UIChart.vue:387]: Gayan, can you clarify why `this.props.xAxisType === 'category' || this.props.xAxisType === 'radial'` was removed from the `if` statement please?

File: ui/src/widgets/ui-chart/UIChart.vue [hunk 2]
```diff
@@ -360,13 +360,9 @@ export default {
...
 
-                // ensure we have a datapoint for the relevant series
-                const data = Array(sLabels.length + 1).fill({})
-                if (xIndex === -1) {
-                    data[xLabels.length] = datapoint
-                } else {
-                    data[xIndex] = datapoint
-                }
+                const data = Array(xLabels.length).fill({}) // initialize the data array
+                data[xIndex] = datapoint
```
Comment (Steve-Mcl) [inline ui/src/widgets/ui-chart/UIChart.vue:364]: what if `xIndex` is `-1` (i.e. not found)?

File: ui/src/widgets/ui-chart/UIChart.vue (hunk 1 above)
Comment (gayanSandamal) [reply ui/src/widgets/ui-chart/UIChart.vue:387]: The condition `this.props.xAxisType === 'category' || this.props.xAxisType === 'radial'` was removed to ensure that the `xIndex >= 0` check applies universally to all axis types. This change makes sure that data points are updated correctly for any axis type, not just category or radial.

File: ui/src/widgets/ui-chart/UIChart.vue (hunk 2 above)
Comment (gayanSandamal) [reply ui/src/widgets/ui-chart/UIChart.vue:364]: Yes, you're correct @Steve-Mcl this should be handled. I'll revisit this and refine

File: ui/src/widgets/ui-chart/UIChart.vue (hunk 1 above)
Comment (gayanSandamal) [reply ui/src/widgets/ui-chart/UIChart.vue:387]: @Steve-Mcl thank you for pointing out. I reverted this line of change as it's not necessary for this change

File: ui/src/widgets/ui-chart/UIChart.vue (hunk 2 above)
Comment (gayanSandamal) [reply ui/src/widgets/ui-chart/UIChart.vue]: Tested this locally with @Steve-Mcl by screen sharing with multiple data points.

Tested chart behaviour manually as this doesn't have E2E tests to check the data visualisation

```
[{"id":"6ef78e61df27712b","type":"group","z":"80a7b66be7ea4735","name":"#1255 - Multi line chart does not draw line for first point on additional topics","style":{"label":true},"nodes":["98571d07252f3e61","3ae1dd08967d2334","35e10b52367f9943","f11b0e84da884431","dac39029fdd65c11","c7b8e797bf25ce1e","78cecaa484d83bb1"],"x":194,"y":2319,"w":732,"h":162},{"id":"98571d07252f3e61","type":"inject","z":"80a7b66be7ea4735","g":"6ef78e61df27712b","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"[{\"topic\":\"t1\",\"payload\":{\"x\":2,\"v\":6},\"_msgid\":\"2279f52ad5738a66\"},{\"topic\":\"t2\",\"payload\":{\"x\":2,\"v\":10},\"_msgid\":\"2279f52ad5738a66\"},{\"topic\":\"t1\",\"payload\":{\"x\":3,\"v\":6},\"_msgid\":\"b59dd85692696016\"},{\"topic\":\"t2\",\"payload\":{\"x\":3,\"v\":10},\"_msgid\":\"b59dd85692696016\"},{\"topic\":\"t1\",\"payload\":{\"x\":4,\"v\":6},\"_msgid\":\"b59dd85692696016\"},{\"topic\":\"t2\",\"payload\":{\"x\":4,\"v\":10},\"_msgid\":\"b59dd85692696016\"}]","payloadType":"json","x":290,"y":2360,"wires":[["3ae1dd08967d2334"]]},{"id":"3ae1dd08967d2334","type":"split","z":"80a7b66be7ea4735","g":"6ef78e61df27712b","name":"","splt":"\\n","spltType":"str","arraySplt":1,"arraySpltType":"len","stream":false,"addname":"","x":470,"y":2360,"wires":[["35e10b52367f9943"]]},{"id":"35e10b52367f9943","type":"change","z":"80a7b66be7ea4735","g":"6ef78e61df27712b","name":"","rules":[{"t":"move","p":"payload.topic","pt":"msg","to":"topic","tot":"msg"},{"t":"set","p":"payload","pt":"msg","to":"payload.payload","tot":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":620,"y":2360,"wires":[["f11b0e84da884431","c7b8e797bf25ce1e"]]},{"id":"f11b0e84da884431","type":"debug","z":"80a7b66be7ea4735","g":"6ef78e61df27712b","name":"debug 2565","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":810,"y":2360,"wires":[]},{"id":"dac39029fdd65c11","type":"inject","z":"80a7b66be7ea4735","g":"6ef78e61df27712b","name":"","props":[{"p":"payload"},{"p":"action","v":"replace","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"[]","payloadType":"json","x":450,"y":2440,"wires":[["c7b8e797bf25ce1e"]]},{"id":"c7b8e797bf25ce1e","type":"ui-chart","z":"80a7b66be7ea4735","g":"6ef78e61df27712b","group":"2eb4d56878d5f814","name":"","label":"chart","order":1,"chartType":"line","category":"topic","categoryType":"msg","xAxisLabel":"x","xAxisProperty":"payload.x","xAxisPropertyType":"msg","xAxisType":"linear","xAxisFormat":"s","xAxisFormatType":"custom","yAxisLabel":"v","yAxisProperty":"v","ymin":"0","ymax":"11","action":"append","stackSeries":false,"pointShape":"circle","pointRadius":4,"showLegend":true,"removeOlder":1,"removeOlderUnit":"3600","removeOlderPoints":"","colors":["#0095ff","#ff0000","#ff7f0e","#2ca02c","#a347e1","#d62728","#ff9896","#9467bd","#c5b0d5"],"textColor":["#666666"],"textColorDefault":true,"gridColor":["#e5e5e5"],"gridColorDefault":true,"width":6,"height":8,"className":"","x":810,"y":2440,"wires":[[]]},{"id":"78cecaa484d83bb1","type":"inject","z":"80a7b66be7ea4735","g":"6ef78e61df27712b","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"[{\"topic\":\"t1\",\"payload\":{\"x\":2,\"v\":6},\"_msgid\":\"2279f52ad5738a66\"},{\"topic\":\"t2\",\"payload\":{\"x\":2,\"v\":10},\"_msgid\":\"2279f52ad5738a66\"}]","payloadType":"json","x":290,"y":2400,"wires":[["3ae1dd08967d2334"]]},{"id":"2eb4d56878d5f814","type":"ui-group","name":"#1255 - Multi line chart does not draw line for first point on additional topics","page":"c2a6a254b088c41d","width":"6","height":"1","order":1,"showTitle":true,"className":"","visible":"true","disabled":"false"},{"id":"c2a6a254b088c41d","type":"ui-page","name":"1111","ui":"a171c8195c1b8e57","path":"/page1111","icon":"wifi","layout":"grid","theme":"0d92c765bfad87e6","order":16,"className":"","visible":"true","disabled":"false"},{"id":"a171c8195c1b8e57","type":"ui-base","name":"My Dashboard","path":"/dashboard","includeClientData":true,"acceptsClientConfig":["ui-control","ui-chart","ui-form","ui-file-input","ui-button","ui-button-group","ui-dropdown","ui-slider","ui-switch","ui-text","ui-table","ui-markdown","ui-notification","ui-template"],"showPathInSidebar":false,"showPageTitle":true,"navigationStyle":"icon","titleBarStyle":"fixed"},{"id":"0d92c765bfad87e6","type":"ui-theme","name":"Basic Blue Theme","colors":{"surface":"#4d58ff","primary":"#0094ce","bgPage":"#eeeeee","groupBg":"#ffffff","groupOutline":"#cccccc"},"sizes":{"pagePadding":"12px","groupGap":"12px","groupBorderRadius":"4px","widgetGap":"12px","density":"default"}}]
```

**Review by Steve-Mcl — APPROVED** [review]
As far as I can see, in pair testing and in my own local testing, this works as expected.

I agree the e2e testing is a huge effort (if even possible) therefore I am happy to approve.

**Pull request merged**